Library/Homebrew/rubocops/python_internal.rb

```diff
@@ -0,0 +1,31 @@
+# typed: true
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits `python-internal` dependencies in formulae.
+      class PythonInternalCop < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, _body_node)
+          dep = depends_on?("python-internal")
+
+          return if dep.blank?
+
+          unless tap_style_exception? :python_internal_allowlist
+            problem "This formula is not allowed to depend on python-internal."
+          end
+
+          dep.child_nodes.each do |child_node|
+            next if !child_node.instance_of?(RuboCop::AST::StrNode) &&
+                    child_node.values.first.child_nodes.length.zero? &&
+                    child_node.values.first.value == :build
```

Comment:
```suggestion
            next if child_node.hash_type? && child_node.values.first.sym_type? &&
                    child_node.values.first.value == :build
```

If not doing the `def_node_matcher` suggestion

Reply:
I went for that solution.